### ISSUE TITLE
Add `match_kinds` explanations with examples in documentations

### ISSUE THREAD

**Issue by heartsker** [realm/SwiftLint#4239]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?
* Installation method used (Homebrew, CocoaPods, building from source, etc)?
* Paste your configuration file:

```yml
# insert yaml contents here
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents.
* Which Xcode version are you using (check `xcodebuild -version`)?
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
// This triggers a violation:
let foo = try! bar()
```

**Comment by SimplyDanny** [comment]
I don't see any useful content in the description. Would you please update/replace it with something helpful?

**Comment by heartsker** [comment]
Yes, I do. For me that was not really clear, e.g. what is the difference between `attribute.builtin` and `attribute.id`?

I would prefer to see one examples for each `match_kind` in the list, just to understand, what is what

I also did not manage finding such info across the Internet

> I don't see any useful content in the description. Would you please update/replace it with something helpful?
